isic/ingest/models/accession.py

```diff
@@ -333,9 +357,115 @@ def metadata_keys():
+            accession_blob = AccessionBlob(
+                blob_name=blob_name,
+                blob=InMemoryUploadedFile(
+                    file=stripped_blob_stream,
+                    field_name=None,
+                    name=blob_name,
+                    content_type=f"image/{output_format.lower()}",
+                    size=stripped_blob_size,
+                    charset=None,
+                ),
+                blob_size=stripped_blob_size,
+                height=img.height,
+                width=img.width,
+                is_cog=False,
+            )
+            self.blob = accession_blob.blob
+            self._accession_blob_field.pre_save(self, add=False)
+
+        return accession_blob
+
+    def _generate_blob_as_cog(self, img: PIL.Image.Image) -> AccessionBlob:
+        with (
+            field_file_to_local_path(self.original_blob) as original_blob_path,
+            tempfile.NamedTemporaryFile(delete=False) as cog_stream,
+        ):
+            gdal.UseExceptions()
+
+            src_ds = gdal.Open(original_blob_path)
+
+            gdal.Translate(
+                cog_stream.name,
+                src_ds,
+                options=gdal.TranslateOptions(
+                    format="COG",
+                    # rescale unsigned 16-bit png band to 8-bit
+                    outputType=gdal.GDT_Byte,
+                    scaleParams=[[0, 2**16 - 1, 0, 2**8 - 1]],
+                    creationOptions={
+                        "BLOCKSIZE": 256,
+                        # "COMPRESS": "JPEG",
+                        # "QUALITY": "10",
+                        "COMPRESS": "DEFLATE",
+                        "PREDICTOR": "YES",
+                        "LEVEL": "9",
+                        "BIGTIFF": "IF_SAFER",
+                        # Strip EXIF metadata
+                        "COPY_SRC_MDD": "NO",
+                    },
+                    resampleAlg=gdal.GRA_Lanczos,
+                ),
+            )
+
+            # necessary to close the src_ds (https://gis.stackexchange.com/a/80370)
+            del src_ds
+
+        with Path(cog_stream.name).open("rb") as cog_stream:
+            blob_name = f"{uuid4()}.tif"
+            blob_size = Path(cog_stream.name).stat().st_size
+            accession_blob = AccessionBlob(
+                blob_name=blob_name,
+                blob=InMemoryUploadedFile(
+                    file=cog_stream,
+                    field_name=None,
+                    name=blob_name,
+                    content_type="image/tiff",
+                    size=blob_size,
+                    charset=None,
+                ),
+                blob_size=Path(cog_stream.name).stat().st_size,
+                height=img.height,
+                width=img.width,
+                is_cog=True,
+            )
+            self.blob = accession_blob.blob
+            self._accession_blob_field.pre_save(self, add=False)
+
+        return accession_blob
```

Comment:
@brianhelba look at this
```suggestion
            yield accession_blob
```